pkg/kubelet/server/remotecommand/exec.go

```diff
@@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
 
 	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
 	if err != nil {
-		msg := fmt.Sprintf("error executing command in container: %v", err)
-		runtime.HandleError(errors.New(msg))
-		fmt.Fprint(ctx.errorStream, msg)
+		m := ExitError{}
```

Comment:
There is but it's done differently. I'll link it when I'm at my desk.

On Thursday, August 4, 2016, Dr. Stefan Schimanski <notifications@github.com>
wrote:

> In pkg/kubelet/server/remotecommand/exec.go
> <https://github.com/kubernetes/kubernetes/pull/26541#discussion_r73508473>
> :
>
> > @@ -51,8 +85,19 @@ func ServeExec(w http.ResponseWriter, req *http.Request, executor Executor, podN
> >
> >  	err := executor.ExecInContainer(podName, uid, container, cmd, ctx.stdinStream, ctx.stdoutStream, ctx.stderrStream, ctx.tty, ctx.resizeChan)
> >  	if err != nil {
> > -		msg := fmt.Sprintf("error executing command in container: %v", err)
> > -		runtime.HandleError(errors.New(msg))
> > -		fmt.Fprint(ctx.errorStream, msg)
> > +		m := ExitError{}
>
> Good point. Old clients would see the JSON and interpret that as the error
> text. I am on the way of refactoring this into a v4 protocol now.
>
> In this context @ncdc <https://github.com/ncdc>: why is there no version
> handshake for the websocket code path?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26541/files/0a10e2dcdaf86a90b5a718e96d44cc9e5286ca06#r73508473>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYuemXq4ZNSgKYwhkg-JPZJX-d2p-ks5qcdOkgaJpZM4Ip8Gm>
> .
>